compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -463,14 +451,28 @@ pub trait PrettyPrinter<'tcx>:
             // `visible_parent_map`), looking for the specific child we currently have and then
             // have access to the re-exported name.
             DefPathData::TypeNs(ref mut name) if Some(visible_parent) != actual_parent => {
-                let reexport = self
+                // Item might be re-exported several times
+                let reexports: Vec<_> = self
                     .tcx()
                     .item_children(visible_parent)
                     .iter()
-                    .find(|child| child.res.opt_def_id() == Some(def_id))
-                    .map(|child| child.ident.name);
-                if let Some(reexport) = reexport {
-                    *name = reexport;
+                    .filter(|child| child.res.opt_def_id() == Some(def_id))
+                    .collect();
+
+                if reexports.len() > 0 {
+                    if let Some(reexport) = reexports
+                        .into_iter()
+                        .find(|child| child.vis.is_public() && child.ident.name != kw::Underscore)
+                        .map(|child| child.ident.name)
+                    {
+                        *name = reexport;
+                    } else {
+                        // There is no name that is public and isn't `_`, so bail.
+                        return Ok((self, false));
+                    }
+                } else {
+                    // FIXME(compiler-errors): What if we don't have an item corresponding to our
```

Comment:
Good question, is there a test case that covers this case?

Reply:
The previous logic didn't do anything, and it's just a missing diagnostic, right? Maybe we could emit a note that the thing is unimportable?

Reply:
Yeah, this is more like the `visible_parent` for the item doesn't actually contain the item at all. This is actually probably a `bug!` (e.g. maybe the `visible_parents_map` has a bug).